x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPainlessExtension.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.spatial;
+
+import org.elasticsearch.painless.spi.PainlessExtension;
+import org.elasticsearch.painless.spi.Whitelist;
+import org.elasticsearch.painless.spi.WhitelistLoader;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.script.ScriptModule;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SpatialPainlessExtension implements PainlessExtension {
+
+    private static final List<Whitelist> WHITELISTS = List.of(
+        WhitelistLoader.loadFromResourceFiles(
+            SpatialPainlessExtension.class,
+            "org.elasticsearch.xpack.spatial.index.fielddata.txt",
```

Comment:
Why split these across three different files?

Reply:
I find it easier to find the classes when they're separated out by package. I realize there's not a huge amount allow-listed here, yet, but I think we want to expand this quite a bit.